app/controllers/addresses_controller.rb

```diff
@@ -1,45 +1,66 @@
 class AddressesController < ApplicationController
+  before_action :find_address, except: [:new, :create]
+
   def new
     @address = Address.new
   end
 
   def create
     @address = Address.new(address_params)
     @address.user_id = session[:user_id]
-    redirect_to home_index_path unless !@address.save
+    if @address.save
+      flash[:notice] = 'The address was succesfully created'
```

Comment:
Why use flash? access the address errors in the view

Reply:
That is not an error

Reply:
It's not, but it smells bad

Reply:
you can iterate flashes in the view with `flash.each` depending of the kind of error you can assign a class to the message using bootstrap https://v4-alpha.getbootstrap.com/components/alerts/